generated_code/NodalBoundaryConditions.py

```diff
@@ -53,7 +53,7 @@ def addKernels(generator, aderdg, include_tensors, matricesDir, dynamicRuptureMe
     # To be used as Tinv in flux solver - this way we can save two rotations for
     # Dirichlet boundary, as ghost cell dofs are already rotated
     identity_rotation = np.double(aderdg.transformation_spp())
-    identity_rotation[0:9, 0:9] = np.eye(9)
+    identity_rotation[0:4, 0:4] = np.eye(4)
```

Comment:
Should be generalized. Maybe use `aderdg.getNumberOfQuantities()` instead of 4 or 9 here in all cases?